required/latex-lab/latex-lab-graphic.dtx

```diff
@@ -137,37 +388,211 @@
+%  It also offers some keys specific to \cs{includegraphics}:
 %  \begin{description}
-%  \item[\texttt{tag}] with the values
+%  \item[\texttt{alt}] This key is already defined in the graphicx package
+%  but redefined here to switch to the illustrative mode and to add its value
+%  as alternative text. 
+%  \item[\texttt{actualtext}] This switches to the actualtext mode.
+%  This is useful for small graphics that represent single chars or a short word like a 
+%  logo. If \texttt{actualtext} is used, the graphics is not enclosed in \texttt{Figure}
+%  structure but in a \texttt{Span} structure and no \texttt{/BBox} attribute is added.%
+%  \footnote{This in accordance with (the draft of) PDF/UA-2 but violates perhaps PDF/UA-1.}
+%  \item[\texttt{artifact}] This tags the graphic as an artifact.
+%  \item[\texttt{tagging-setup}] This key takes as argument a key-list,
+%  which can contain the following keys:
 %   \begin{description}
-%    \item[\texttt{artifact}] When used the graphic will be tagged as artifact. This doesn't
-%     require a \texttt{BBox} and so works also in some of the not yet supported cases described
-%     above.
-%   \item[\texttt{false}] When used tagging will be stopped completely. It is then the 
+%    \item[\texttt{alt=\meta{text}}] This a second way to tag the graphic as figure with
+%    alternative text. \texttt{alt} without value will tag as figure but not change
+%    the text variable, \texttt{alt=} will empty the text variable (and typically trigger
+%    a warning).
+%    \item[\texttt{actualtext=\meta{text}}] This a second way to tag the graphic as a
+%    symbol with actualtext.  
+%    \item[\texttt{artifact}] This a second way to tag the graphic as artifact. 
+%    \item[\texttt{text}] This switches to text mode.
+%    \item[\texttt{off}] When used tagging will be stopped completely. It is then the 
 %   responsibility of the surrounding code to add appropriate tagging commands.
-%   \item[\meta{name}] Other values will be used as tag names in the structure. If the tag is not
-%   known as a structure tag you will get an warning from tagpdf. The default name is currently
-%   \texttt{Figure}
+%   \item[\texttt{tag=\meta{name}}] This switches to the illustrative mode but uses
+%   \meta{name} as tag name in the structure instead of the default \texttt{Figure}. 
+%   This can for example be used to tag an image of a formula with \texttt{Formula}.
 %   \end{description}   
-%  \item[\texttt{actualtext}] This allows to add an /ActualText to the structure. 
-%  This is useful for small graphics that represent single chars or a short word like a 
-%  logo. If \texttt{actualtext} is used, the graphics is not enclosed in \texttt{Figure}
-%  structure but in a \texttt{Span} structure and no \texttt{/BBox} attribute is added.
-%  This in accordance with (the draft of) PDF/UA-2 but violates perhaps PDF/UA-1.
 %  \item[\texttt{correct-BBox}] If the calculated \texttt{/BBox} values are wrong
```

Comment:
What (might) be useful here is an ability to specify the BBox relative to the reference point of the figure. If the image is generated externally you have control over its natural size, but less control over where ex puts it on the page.

Actually I think the name is bad I originally read `correct` as an adjuctive that is, this is supplying the correct values, but  if I read the code correctly this is `BoundingBox-Correction` that is an instruction to amend the values by the specified amount. I still think that is a bit fragile as if the original values are wrong as the natural size is mis-calculated, the correction is stable between edits to the document, but if the adjustment is due to the actual position on the page, then they are not.

Reply:
The correction is relative to the calculation done by the code, so if you see (with debug=BBox) that a centimeter is missing on the left you give this centimeter as correction. I don't think that adjustments or sizes relative to the reference point to the figure is really useful. If an image is simply inserted, the automatic calculation should be fine, the problems start if there is lots of (external) resizing and rotation is going on, and then it is difficult to know where the reference point is. On the whole I hope that key is seldom needed ...